### ISSUE TITLE
[deliver] Keeps asking for password using FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD

### ISSUE THREAD

**Issue by Maxwell2022** [fastlane/fastlane#14693]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I just want to upload a binary to TestFlight.

I read all the posts that were saying "do no update metadata, etc..". What I want to do is just upload the binary. I've set `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD` with a password I've generated in the apple account page.

I've also tried to set every option to `false` even tho it supposed to be the default value. I'm also using the docker image (`fastlanetools/fastlane`) to make sure it's not using my keychain values.

This is the lane in question:

```
desc 'Test Deliver'
  lane :deliver_test do
    deliver(
      ipa: ***********,
      force: true, 
      skip_screenshots: true,
      skip_metadata: true,
      username: ***********,
      app_identifier: *********,
      team_id: '************',
      run_precheck_before_submit: false,
      edit_live: false,
      use_live_version: false,
      skip_binary_upload: false,
      skip_app_version_update: false,
      overwrite_screenshots: false,
      submit_for_review: false,
      reject_if_possible: false,
      automatic_release: false, 
      phased_release: false,
      ignore_language_directory_validation: true
    )
  end
```

##### Command executed

```
bundle exec fastlane ios deliver_test
```

##### Complete output when running fastlane, including the stack trace and command used

```
bundle exec fastlane ios deliver_test
$ bundle exec fastlane ios deliver_test
[✔] 🚀
[03:03:32]: Driving the lane 'ios deliver_test' 🚀
[03:03:32]: ---------------------------
[03:03:32]: --- Step: opt_out_usage ---
[03:03:32]: ---------------------------
[03:03:32]: Disabled upload of used actions
[03:03:32]: ---------------------
[03:03:32]: --- Step: deliver ---
[03:03:32]: ---------------------
[03:03:32]: Login to App Store Connect (********@********.com.au)
-------------------------------------------------------------------------------------
Please provide your Apple Developer Program account credentials
You can also pass the password using the `FASTLANE_PASSWORD` environment variable
-------------------------------------------------------------------------------------
Password (for ********@********.com.au):
```

### Environment

running env:

```

$ env
RUBY_MAJOR=2.6
HOSTNAME=44bfdd18f078
FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD=****-****-****-****
HOME=/home/circleci
BUNDLE_APP_CONFIG=/usr/local/bundle
JAVA_VERSION=8u131
RUBY_VERSION=2.6.2
CA_CERTIFICATES_JAVA_VERSION=20161107~bpo8+1
TERM=xterm
JAVA_DEBIAN_VERSION=8u131-b11-1~bpo8+1
PATH=/usr/local/bundle/bin:/usr/local/bundle/gems/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/itms/bin
BUNDLE_PATH=/usr/local/bundle
LANG=C.UTF-8
DEBIAN_FRONTEND=noninteractive
GEM_HOME=/usr/local/bundle
RUBY_DOWNLOAD_SHA256=91fcde77eea8e6206d775a48ac58450afe4883af1a42e5b358320beb33a445fa
LC_ALL=C.UTF-8
JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/jre
PWD=/app
BUNDLE_SILENCE_ROOT_WARNING=1
```

I'm using the docker image so it should be pretty standard

<details>
  <pre>
[✔] 🚀
[02:58:02]: Generating fastlane environment output, this might take a few seconds...
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                  |
| --------------------------- | ------------------------------------------------------ |
| OS                          | debian_based 9.8                                       |
| Ruby                        | 2.6.2                                                  |
| Bundler?                    | true                                                   |
| Git                         | git version 2.11.0                                     |
| Installation Source         | /usr/local/bundle/bin/fastlane                         |
| Host                        | Debian GNU/Linux 9 debian_based 9.8 (4.9.125-linuxkit) |
| Ruby Lib Dir                | /usr/local/lib                                         |
| OpenSSL Version             | OpenSSL 1.1.0j  20 Nov 2018                            |
| Is contained                | false                                                  |
| Is homebrew                 | false                                                  |
| Is installed via Fabric.app | false                                                  |

### System Locale

| Variable | Value   |   |
| -------- | ------- | - |
| LANG     | C.UTF-8 | ✅ |
| LC_ALL   | C.UTF-8 | ✅ |
| LANGUAGE |         |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
before_all do
  opt_out_usage
  update_fastlane
end

platform :ios do
  # iOS Lanes
  desc 'Fetch certificates and provisioning profiles'
  private_lane :certificates do
    match(app_identifier: 'com.**********.app', type: 'development', readonly: true, username: ENV["MATCH_USERNAME"])
    match(app_identifier: 'com.**********.app', type: 'appstore', readonly: true, username: ENV["MATCH_USERNAME"])
  end

  desc '[CI] Build the iOS application'
  lane :ci_build do
    setup_circle_ci
    certificates
    gym(scheme: 'MyAwesomeProject', workspace: './ios/MyAwesomeProject.xcworkspace')
  end

  desc 'Build the iOS application'
  lane :build do
    certificates
    gym(scheme: 'MyAwesomeProject', workspace: './ios/MyAwesomeProject.xcworkspace')
  end

  desc 'Ship to Testflight'
  lane :beta do
    begin
      setup_circle_ci
      ensure_git_branch(branch: 'develop')
      ensure_git_status_clean
      increment_build_number(xcodeproj: './ios/MyAwesomeProject.xcodeproj')
      build(scheme: 'MyAwesomeProject', workspace: './ios/MyAwesomeProject.xcworkspace')
      commit_version_bump(message: 'Bump build number', xcodeproj: './ios/MyAwesomeProject.xcodeproj')
      push_to_git_remote
      deliver(
        ipa: './output/gym/MyAwesomeProject.ipa',
        force: true, # Skip HTMl report verification
        skip_screenshots: true,
        skip_metadata: true,
        username: ENV["MATCH_USERNAME"],
        app_identifier: 'com.**********.app',
        team_id: '************',
        run_precheck_before_submit: false,
        edit_live: false,
        use_live_version: false,
        skip_binary_upload: false,
        skip_app_version_update: false,
        overwrite_screenshots: false,
        submit_for_review: false,
        eject_if_possible: false,
        automatic_release: false,
        phased_release: false,
        ignore_language_directory_validation: false
      )
    rescue => exception
      on_error(exception, lane_context[SharedValues::LANE_NAME])
    end
  end

  desc 'Test Deliver'
  lane :deliver_test do
    deliver(
      ipa: './ios/output/gym/MyAwesomeProject.ipa',
      force: true, # Skip HTMl report verification
      skip_screenshots: true,
      skip_metadata: true,
      username: ENV["MATCH_USERNAME"],
      app_identifier: 'com.**********.app',
      team_id: '************',
      run_precheck_before_submit: false,
      edit_live: false,
      use_live_version: false,
      skip_binary_upload: false,
      skip_app_version_update: false,
      overwrite_screenshots: false,
      submit_for_review: false,
      reject_if_possible: false,
      automatic_release: false,
      phased_release: false,
      ignore_language_directory_validation: true
    )
  end
end



def on_error(exception, lane_name)
  slack(
      message: "Something went wrong (#{lane_name})",
      success: false,
      channel: '#MyAwesomeProject',
      attachment_properties: {
          fields: [
              {
                  title: "Lane",
                  value: lane_name,
              },
              {
                  title: "Error message",
                  value: exception.to_s,
                  short: false
              }
          ]
      }
  )
end
```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.121.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| bundler                   | 1.17.2       |
| etc                       | 1.0.1        |
| forwardable               | 1.2.0        |
| CFPropertyList            | 3.0.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.6.0        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.4.1        |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| dotenv                    | 2.7.2        |
| emoji_regex               | 1.0.1        |
| excon                     | 0.64.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.15.4       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.5        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| multi_json                | 1.13.1       |
| os                        | 1.0.1        |
| signet                    | 0.11.0       |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2019.0331  |
| mime-types                | 3.2.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.0.5        |
| google-cloud-core         | 1.3.0        |
| google-cloud-storage      | 1.16.0       |
| json                      | 2.2.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.5.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.5.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.5        |
| tty-cursor                | 0.6.1        |
| tty-spinner               | 0.9.0        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.8.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
</details>

*generated on:* **2019-04-30**
</details>
</pre>
</details>

**Comment by janpio** [comment]
https://github.com/fastlane/fastlane/pull/14456 has to be merged first for a way to _not_ getting asked for your normal password when using deliver or pilot to upload a binary only.

**Comment by Maxwell2022** [comment]
@janpio Is there any alternative at the moment that is not using spaceauth and sessions?

**Comment by janpio** [comment]
No.

You can of course use an account that does not have 2FA enable and then set `FASTLANE_PASSWORD`, but you didn't mention your motivation so I don't know if this is your reason to try.

**Comment by Maxwell2022** [comment]
> can of course use an account that does not

Thanks @janpio I actually thought it was compulsory to have the 2FA on all the account, however, it's just for the main account. I've created a CI user without 2FA and it's all working nicely now
